### ISSUE TITLE
Request: Add alias storedir for store-dir

### ISSUE THREAD

**Issue by AndreasPK** [haskell/cabal#5484]
Given that options specifying build output paths usually use --<X>dir (builddir, bindir, ...) that seems reasonable.

**Comment by hvr** [comment]
maybe do it the other way around; rename `store-dir` to `storedir`, and add a (deprecated) `store-dir` aliasing `storedir`

cc'ing @alexbiehl who implemented `--store-dir`

**Comment by 23Skidoo** [comment]
Don't think we need to deprecate the alias, it's easier to keep supporting it.

**Comment by ChShersh** [comment]
@hvr @23Skidoo Could you, please, provide some guidance regarding this issue?

I found this place in `cabal-install` where `store-dir` command-line option is mentioned:

```
cabal-install/Distribution/Client/Setup.hs
449:      ,option [] ["store-dir"]
```

Am I right that I just need to add the following to `argsNotShown`:

```haskell
      ,option [] ["storedir"]
         "The location of the nix-local-build store"
         globalStoreDir (\v flags -> flags { globalStoreDir = v })
         (reqArgFlag "DIR")
```

Or am I'm missing some mechanism for aliasing?

**Comment by 23Skidoo** [comment]
I think you can just add `"storedir"` to the `["store-dir"]` list.

**Comment by ChShersh** [comment]
Fixed in #5527
